### ISSUE TITLE
4.2.1 does not work with Java 22 EA builds

### ISSUE THREAD

**Issue by wilkinsona** [awaitility/awaitility#275]
We've recently upgraded Spring Boot to Awaitility 4.2.1 from 4.2.0 and our Java 22 CI builds are now failing with the following exception:

```
java.lang.NumberFormatException: For input string: "22-beta"
    at java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
    at java.lang.Integer.parseInt(Integer.java:588)
    at java.lang.Integer.parseInt(Integer.java:685)
    at org.awaitility.core.JavaVersionDetector.getJavaMajorVersion(JavaVersionDetector.java:21)
    at org.awaitility.core.LambdaErrorMessageGenerator.getLambdaDetectionClassName(LambdaErrorMessageGenerator.java:86)
    at org.awaitility.core.LambdaErrorMessageGenerator.isLambdaClass(LambdaErrorMessageGenerator.java:30)
    at org.awaitility.core.CallableHamcrestCondition.getCallableDescription(CallableHamcrestCondition.java:51)
    at org.awaitility.core.AbstractHamcrestCondition.getMatchMessage(AbstractHamcrestCondition.java:68)
    at org.awaitility.core.AbstractHamcrestCondition.lambda$new$0(AbstractHamcrestCondition.java:51)
    at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:248)
    at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:235)
    at java.util.concurrent.FutureTask.run(FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    at java.lang.Thread.run(Thread.java:1570
```

We'll roll back to 4.2.0 until Java 22 has GAed later this week.

**Comment by yrodiere** [comment]
We're getting the same problem on Hibernate Search with JDK 23 EA:

```
java.lang.NumberFormatException: For input string: "23-ea"
    at java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
    at java.lang.Integer.parseInt(Integer.java:588)
    at java.lang.Integer.parseInt(Integer.java:685)
    at org.awaitility.core.JavaVersionDetector.getJavaMajorVersion(JavaVersionDetector.java:21)
```

**Comment by yrodiere** [comment]
See also #277

**Comment by snicoll** [comment]
Should be fixed by #279
